libs/knowledge-store/ragstack_knowledge_store/directed_edge_extractor.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+from typing import Any, Dict, Iterable
+from ragstack_knowledge_store.edge_extractor import EdgeExtractor
+from ragstack_knowledge_store.knowledge_store import KnowledgeStore
+
+
+class DirectedEdgeExtractor(EdgeExtractor):
+    """Extract edges between uses (of a URL or other ID) and definitions (of a URL or other ID).
+
+    While `UndirectedEdgeExtractor` links nodes in both directions if they share
+    a keyword, this only creates links from nodes with a "source" to nodes with
+    a matching "target". For example, uses may be the `href` of `a` tags in the
+    chunk and definitions may be the URLs that the chunk is accessible at.
+
+    This may also be used for other forms of references, such as Wikipedia article IDs, etc.
+    """
+
+    def __init__(self,
+                 sources_field: str,
```

Comment:
This only partly makes these fields configurable -- note the `KnowledgeStore` is currently still responsible for writing metadata from these fields, so it's not perfect yet. 

Ideally, either the `KnowledgeStore` would consult the extractors (to figure out what columns to populate an how) which could get tricky / tightly coupled or each extractor would be responsible for it's own writes (which could be annoying since there'd be `K+1` writes for each node with `K` extractors).